trunk/src/meshmod/nodmod.F90

```diff
@@ -112,6 +112,7 @@ subroutine nodmod(Nod,Nordn)
                deallocate(NODES(Nod)%dof%zdofH)
             endif
          endif
+!$omp atomic
```

Comment:
What is the performance penalty on this? What if instead of atomically updating in here (though we can leave it if someone happens to call if from within OMP parallel region), we make a second version of the routine, nodmod_omp which returns the change value to each NRDOFSH,E,V,Q. This can be accumulated independently by each OMP thread calling nodmod_omp, and then reduced and added to the global variables at the very end.

Reply:
I haven't checked that; I'd doubt it's very high though (I haven't notices any significant slowdown), especially relative to other things like allocating DoFs. I'm leaving for ABQ soon; perhaps we can merge this as is and make an issue to look into it in the future? It hasn't come up on any of my past profiles and I made this change a while ago so it's unlikely to be a bottleneck.

Reply:
sure we can merge it as is and test very easily if there's a performance difference. Atomic ops can cause a significant slow-down but you're right it's probably dwarfed by the dof allocations.